packages/typespec-ts/src/index.ts

```diff
@@ -52,142 +48,200 @@ import { buildSubpathIndexFile } from "./modular/buildSubpathIndex.js";
+  const dpgContext = createSdkContext(context) as SdkContext;
   const needUnexpectedHelper: Map<string, boolean> = new Map<string, boolean>();
   const serviceNameToRlcModelsMap: Map<string, RLCModel> = new Map<
     string,
     RLCModel
   >();
-  for (const client of clients) {
-    count++;
-    const rlcModels = await transformRLCModel(
-      program,
-      options,
-      client,
-      context.emitterOutputDir,
-      dpgContext
-    );
-    serviceNameToRlcModelsMap.set(client.service.name, rlcModels);
-    const pathToClear = rlcModels.srcPath;
-    needUnexpectedHelper.set(client.name, hasUnexpectedHelper(rlcModels));
-    clearSrcFolder(
-      pathToClear,
-      count,
-      options.multiClient,
-      options.isModularLibrary
-    );
-    await emitModels(rlcModels, program);
-    await emitContentByBuilder(program, buildClientDefinitions, rlcModels);
-    await emitContentByBuilder(program, buildResponseTypes, rlcModels);
-    await emitContentByBuilder(program, buildClient, rlcModels);
-    await emitContentByBuilder(program, buildParameterTypes, rlcModels);
-    await emitContentByBuilder(program, buildIsUnexpectedHelper, rlcModels);
-    await emitContentByBuilder(program, buildIndexFile, rlcModels);
-    await emitContentByBuilder(program, buildLogger, rlcModels);
-    await emitContentByBuilder(program, buildTopLevelIndex, rlcModels);
-    await emitContentByBuilder(program, buildPaginateHelper, rlcModels);
-    await emitContentByBuilder(program, buildPollingHelper, rlcModels);
-    // buildSerializeHelper
-    await emitContentByBuilder(program, buildSerializeHelper, rlcModels);
-    // build metadata relevant files
-    await emitContentByBuilder(
-      program,
-      [
-        buildEsLintConfig,
-        buildRollupConfig,
-        buildApiExtractorConfig,
-        buildReadmeFile,
-        buildPackageFile,
-        buildTsConfig
-      ],
-      rlcModels,
-      context.emitterOutputDir
-    );
-    // build test relevant files
-    await emitContentByBuilder(
-      program,
-      [
-        buildKarmaConfigFile,
-        buildEnvFile,
-        buildEnvBrowserFile,
-        buildRecordedClientFile,
-        buildSampleTest
-      ],
-      rlcModels,
-      context.emitterOutputDir
-    );
+  const generationPathDetail: GenerationDirDetail =
+    await calculateGenerationDir();
+  dpgContext.generationPathDetail = generationPathDetail;
+  // 1. Clear sources folder
+  clearSrcFolder();
+  // 2. Generate RLC sources
+  await generateRLC();
+  // 3. Generate Modular sources
+  await generateModular();
+
+  async function calculateGenerationDir(): Promise<GenerationDirDetail> {
+    const projectRoot = context.emitterOutputDir ?? "";
+    let sourcesRoot = join(projectRoot, "src");
+    const customizationFolder = join(projectRoot, "sources");
+    if (await fsextra.pathExists(customizationFolder)) {
+      sourcesRoot = join(customizationFolder, "generated", "src");
```

Comment:
Have we decided to use `sources/generated/src` folder?

Reply:
This is the way of current openai implementation and customization tool supported. Personally I think `sources/generated` and `sources/generated/src` would work me. The latter one could give us more flexibility if in future we'd like to add more metadata or testing files.

@joheredi How do you think?